node_test.go

```diff
@@ -0,0 +1,97 @@
+package substrate
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	validNode = Node{
+		Versioned: Versioned{
+			Version: 5,
+		},
+		ID:     14,
+		FarmID: 1,
+		TwinID: 22,
+		Resources: Resources{
+			HRU: 9001778946048,
+			SRU: 512110190592,
+			CRU: 24,
+			MRU: 202802929664,
+		},
+		Location: Location{
+			Longitude: "4.3447",
+			Latitude:  "50.8509",
+		},
+		Country: "Belgium",
+		City:    "Unknown",
+		PublicConfig: OptionPublicConfig{
+			HasValue: true,
+			AsValue: PublicConfig{
+				IP4: IP{
+					IP: "185.206.122.31/24",
+					GW: "185.206.122.1",
+				},
+				IP6: OptionIP{
+					HasValue: true,
+					AsValue: IP{
+						IP: "2a10:b600:1::0cc4:7a30:65b5/64",
+						GW: "2a10:b600:1::1",
+					},
+				},
+				Domain: OptionDomain{
+					HasValue: true,
+					AsValue:  "gent01.dev.grid.tf",
+				},
+			},
+		},
+		Created:       1649252280,
+		FarmingPolicy: 1,
+		Interfaces: []Interface{
+			{
+				Name: "zos",
+				Mac:  "0c:c4:7a:30:7a:68",
+				IPs:  []string{"10.9.0.113", "2a10:b600:0:9:ec4:7aff:fe30:7a68"},
+			},
+		},
+		Certification: NodeCertification{
+			IsDiy:       true,
+			IsCertified: false,
+		},
+		BoardSerial:     "NM141S001819",
+		ConnectionPrice: 800,
+	}
+)
+
+func TestGetNodeByTwinID(t *testing.T) {
+	cl := startConnection(t)
+	defer cl.Close()
+
+	nodeID, err := cl.GetNodeByTwinID(uint32(validNode.TwinID))
+
+	require.NoError(t, err)
+	require.Equal(t, uint32(validNode.ID), nodeID)
+}
+
+func TestGetNode(t *testing.T) {
+
+	cl := startConnection(t)
+	defer cl.Close()
+
+	node, err := cl.GetNode(uint32(validNode.ID))
+
+	require.NoError(t, err)
+	require.Equal(t, &validNode, node)
+}
+
+func TestGetLastNodeID(t *testing.T) {
+
+	cl := startConnection(t)
+	defer cl.Close()
+
+	nodeID, err := cl.GetLastNodeID()
+
+	require.NoError(t, err)
+	require.Equal(t, uint32(49), nodeID)
+}
```

Comment:
This is an invalid test unfortunately, because this will fail on the next node join devnet because the last node id will change to 50. so please drop

Reply:
I removed this test, but I didn't commit it yet